tests/xrt/import_bo/import.cpp

```diff
@@ -0,0 +1,186 @@
+#include <unistd.h>
+#include <cstring>
+#include <iostream>
+#include <chrono>
+#include <thread>
+
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_kernel.h"
+
+// Test export / import of XRT buffer object between processes.
+//
+// Test requires pidfd Linux kernel support and is supposed only on
+// x86.  The example has been tested on Ubuntu 21.10
+//
+// % uname -r
+// 5.13.0-22-generic
+//
+// Also, note that the importing process must have permission to
+// duplicate the exporting process' file descriptor.  This permission
+// is controlled by ptrace access mode PTRACE_MODE_ATTACH_REALCREDS
+// check (see ptrace(2)).  Alternatively, run the example as root.
+//
+// In the test, the parent process writes "parent" to the BO. The
+// child child process waits for the host buffer to contain the parent
+// string and writes "child" and program terminates after parent sees
+// the child string.
+//
+// The program allocates the buffer in memory bank 0, so make sure a
+// corresponding xclbin is loaded, e.g. bandwidth.xclbin
+//
+// % g++ -g -std=c++14 -I${XILINX_XRT}/include -L${XILINX_XRT}/lib -o import.exe import.cpp -lxrt_coreutil -pthread -luuid
+//
+// # kernel allocates host buffer
+// % import.exe -k verify.xclbin
+//
+// # userspace allocates host buffer (fails in xrt-2.12.x)
+// % import.exe -k verify.xclbin --ubuf
+
+// Golden pattern from kernel
+static constexpr char gold[] = "Hello World\n";
+
+// A pipe to communicate exported bo handle to child
+static int talk[2];
```

Comment:
Magic number.  What is so special about 2 :^)

Reply:
It's not magic, it is a linux pipe, man pipe https://man7.org/linux/man-pages/man2/pipe.2.html